src/geom/segment.jl

```diff
@@ -0,0 +1,117 @@
+
+abstract SegmentGeometry <: Gadfly.GeometryElement
+
+# Geometry for vectors/arrows/segments
+immutable SegmentGeom <: SegmentGeometry 
+    
+     arrow::Bool
+     filled::Bool
+     tag::Symbol 
+ 
+    function SegmentGeom(;arrow=false, filled=false, tag=empty_tag) 
+         new(arrow, filled, tag) 
+     end 
+end 
+
+const segment = SegmentGeom
+
+# Leave this as a function, pending extra arguments e.g. arrow attributes
+function vector(;filled::Bool=false, tag::Symbol=empty_tag)
+    return SegmentGeom(arrow=true, filled=filled, tag=tag)
+end
+
+ 
+function element_aesthetics(::SegmentGeom) 
+    return [:x, :y, :xend, :yend, :color] 
+end 
+
+
+function Gadfly.render(geom::SegmentGeom, theme::Gadfly.Theme, aes::Gadfly.Aesthetics,
+                subplot_layer_aess::@compat(Union{(@compat Void), Vector{Gadfly.Aesthetics}}),
+                subplot_layer_datas::@compat(Union{(@compat Void), Vector{Gadfly.Data}}),
+    scales::Dict{Symbol, Gadfly.ScaleElement})
+    render(geom, theme, aes, scales)
+end
+
+
+
+
+
+function render(geom::SegmentGeom, theme::Gadfly.Theme, aes::Gadfly.Aesthetics,
+    scales::Dict{Symbol, Gadfly.ScaleElement})
+
+    Gadfly.assert_aesthetics_defined("Geom.segment", aes, :x, :y, :xend, :yend)
+
+    function arrow{T<:Real}(x::T, y::T, xmax::T, ymax::T, xyrange::Vector{T})
+        dx = xmax-x
+        dy = ymax-y
+        vl = 0.225*hypot(dy/xyrange[2], dx/xyrange[1])
+        θ =  atan2(dy/xyrange[2], dx/xyrange[1])
+        ϕ = pi/15
+        xr = -vl*xyrange[1]*[cos(θ+ϕ), cos(θ-ϕ)]
+        yr = -vl*xyrange[2]*[sin(θ+ϕ), sin(θ-ϕ)]
+        arr = [(xmax+xr[1],ymax+yr[1]), (xmax,ymax), (xmax+xr[2],ymax+yr[2]) ]
+        return arr
+    end
+
+    n = length(aes.x)
+    color = ColorTypes.RGBA{Float32}(theme.default_color)
+    default_aes = Gadfly.Aesthetics()  
+    default_aes.color = DataArrays.DataArray(fill(color,n))
+
+    aes = inherit(aes, default_aes) 
+
+#    line_style = Gadfly.get_stroke_vector(theme.line_style)
+    line_style = theme.line_style 
+    if is(line_style, nothing)
+        line_style = [] 
+    end
+
+
+# Geom.vector requires information about scales
+
+if geom.arrow
```

Comment:
Is the top-level if statement necessary because `render` doesn't get scale information right now? Because this looks a bit hacky.